docs/user-guide/installation.rst

```diff
@@ -1,9 +1,23 @@
 Installation
 ============
 
+Required dependencies
+---------------------
+
+Required dependencies include:
+
+- `Python <https://docs.python.org/3/>`_ (3.11 or later)
+- `packaging <https://packaging.pypa.io>`_ (22.0 or later)
+- `numpy <https://numpy.org>`_ (1.25 or later)
+- `numcodecs[crc32c] <https://numcodecs.readthedocs.io>`_ (0.14 or later)
+- `typing_extensions <https://typing-extensions.readthedocs.io>`_ (4.9 or later)
+- `donfig <https://donfig.readthedocs.io>`_ (0.8 or later)
```

Comment:
Listing these here is one more place they need updating when we update any of these versions or dependencies. Since pip and conda automatically handle dependencies, I'd be pro getting rid of this section entirely?